### PR TITLE
Merge develop to main

### PR DESCRIPTION
**Pull request by karisal-anders**

### CODE REVIEW

File: backend/tet/tet/views.py [hunk 1]
```diff
@@ -104,8 +106,16 @@
     authentication_classes = [ApiTokenAuthentication]
     permission_classes = [GDPRScopesPermission]
 
+    def dispatch(self, request, *args, **kwargs):
+        LOGGER.warning(
+            f'TetGDPRAPIView auth="{request.META.get("HTTP_AUTHORIZATION")}"'
+        )  # DEBUG!
```
[the remark this answers is not in the document]
Comment (karisal-anders) [reply backend/tet/tet/views.py:112]: https://github.com/City-of-Helsinki/yjdh/pull/1635#discussion_r1055169944 i.e.
> This is noted, after discussion we agreed (@karisal-anders and @nenonja) that this is acceptable if not published to production environment at all and removed after debugging has been done.

So not releasing this version to production on Platta, trying this out on staging.

File: backend/tet/requirements.in [hunk 2]
```diff
@@ -9,7 +9,7 @@ django-simple-history
 django-storages[azure]
 djangorestframework
 django~=3.2
-drf-oidc-auth
+drf-oidc-auth==0.10.0  # For GDPR API to work, otherwise got 'Invalid claim: "amr"'
```
Comment (nikomakela) [inline backend/tet/requirements.in:12]: This is almost 3 years old and not even a major version. I do understand the issue there, but a authentication is such a critical area where I would follow the latest major versions when there are those available.

In Kultus and Kukkuu there is an alternative fix for this, which converts the amr to list when it is not:
https://github.com/City-of-Helsinki/palvelutarjotin/blob/master/palvelutarjotin/oidc.py#L4. It should be easy and simple to activate in this project too.

Comment (karisal-anders) [reply backend/tet/requirements.in:12]: Ok, sounds good to fix before production release. I will change this before releasing to production.

**Review by nikomakela — APPROVED** [review]
Approved only for the test use, since the author does not have admin privileges to bypass the approval process 😄 I really would recommend to update the drf-oidc-auth -lib to the latest.

**Pull request merged**